### PR TITLE
Seabeta 444 Add epc hint to property age

### PR DESCRIPTION
**Pull request by CodrinCotarlanSoftwire**

### CODE REVIEW

File: SeaPublicWebsite/Views/EnergyEfficiency/HomeAge.cshtml [hunk 1]
```diff
@@ -24,56 +25,80 @@
...
+                            Your @Model.Epc?.LodgementDate?.Year Energy Performance Certificate suggests your property was built
+                            <strong>
+                                @(Model.Epc?.ConstructionAgeBand switch {
+                                    HomeAge.Pre1900 or HomeAge.From1900To1929 => "before 1900.",
+                                    HomeAge.From1930To1949 or HomeAge.From1950To1966 => "between 1930 and 1966.",
+                                    HomeAge.From1967To1975 or HomeAge.From1976To1982 => "between 1967 and 1982.",
+                                    HomeAge.From1983To1990 or HomeAge.From1991To1995 => "between 1983 and 1995.",
+                                    HomeAge.From1996To2002 or HomeAge.From2003To2006 => "between 1996 and 2011.",
+                                    HomeAge.From2007ToPresent => "on or after 2012.",
+                                    _ => throw new ArgumentOutOfRangeException()})
```
Comment (CodrinCotarlanSoftwire) [inline SeaPublicWebsite/Views/EnergyEfficiency/HomeAge.cshtml:59]: Recommendation service uses this inference logic:
`HomeAge.From2003To2006 => "J",
                    HomeAge.From2007ToPresent => "L",
                    _ => wallConstruction switch`
i.e. epcs for buildings newer than 2007 are treated as band L (i.e.  after 2012

File: SeaPublicWebsite/ExternalServices/EpbEpc/EpbEpcApi.cs [hunk 2]
```diff
@@ -486,21 +486,75 @@ private async Task<string> RequestTokenIfNeeded()
...
+            if (epc.PropertyAgeBand.Equals("I", StringComparison.OrdinalIgnoreCase) ||
+                epc.PropertyAgeBand.Contains("1996-2002"))
+            {
+                return HomeAge.From1996To2002;
+            }
+            
+            if (epc.PropertyAgeBand.Equals("J", StringComparison.OrdinalIgnoreCase) ||
+                epc.PropertyAgeBand.Contains("2003-2006"))
+            {
+                return HomeAge.From1900To1929;
```
Comment (DanCorderSoftwire) [inline SeaPublicWebsite/ExternalServices/EpbEpc/EpbEpcApi.cs:546]: This looks suspicious

File: SeaPublicWebsite/Views/EnergyEfficiency/HomeAge.cshtml (hunk 1 above)
Comment (DanCorderSoftwire) [reply SeaPublicWebsite/Views/EnergyEfficiency/HomeAge.cshtml:59]: Hmm, that may be necessary but it sounds suspicious. Can you check that Joanna is happy with this?

Comment (DanCorderSoftwire) [reply SeaPublicWebsite/Views/EnergyEfficiency/HomeAge.cshtml:59]: Separately, could you put this logic into the view model please - we should start getting into the habit of putting minimal logic into the view itself (view models are much more easily testable)

**Review by DanCorderSoftwire — CHANGES_REQUESTED** [review]

**Comment by CodrinCotarlanSoftwire** [comment]
Captured them both in a ticket, will follow them up.

**Pull request merged**